zenoh-flow/src/io/input.rs

```diff
@@ -259,10 +280,18 @@ impl<T: ZFData + 'static> Input<T> {
     /// Note that if some channels are disconnected, for each of such channel an error is logged.
     pub fn try_recv(&self) -> Result<(Message<T>, Timestamp)> {
         match self.input_raw.try_recv()? {
-            LinkMessage::Data(DataMessage { data, timestamp }) => {
-                Ok((Message::Data(Data::try_new(data)?), timestamp))
-            }
+            LinkMessage::Data(DataMessage { data, timestamp }) => Ok((
+                Message::Data(Data::try_from_payload(
+                    data,
+                    Arc::clone(&self.deserializer),
```

Comment:
Why `Arc::clone`? Few lines before it is just `clone`, is the compiler unable to understand that we want to clone the `Arc` and not the content?